.github/workflows/ci.yaml

```diff
@@ -109,15 +111,26 @@ jobs:
         restore-keys: |
           ${{ runner.os }}-go-
     - name: Install buf
-      run: go install github.com/bufbuild/buf/cmd/buf@v1.26.1
-    - name: Install controller-gen
-      run: go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.14.0
-    - name: Run Codegen
-      run: make codegen
+      uses: bufbuild/buf-setup-action@v1.29.0-1
+      with:
+        version: 1.26.1
+    - name: Install Protoc
+      uses: arduino/setup-protoc@v3
+      with:
+        version: "25.3"
+    - name: Install linter
+      env:
+        GOLANGCI_LINT_VERSION: 1.54.2
+      working-directory: /usr/local/bin
+      run: |
+        curl -sSfL https://github.com/golangci/golangci-lint/releases/download/v${GOLANGCI_LINT_VERSION}/golangci-lint-${GOLANGCI_LINT_VERSION}-linux-amd64.tar.gz \
+        | tar xvz golangci-lint-${GOLANGCI_LINT_VERSION}-linux-amd64/golangci-lint --strip-components=1
```

Comment:
I'm curious why we need this linter here.

Reply:
I used it for formatting updated Kubebuilder code. Changed to using `go/format` in `prototag` util in https://github.com/akuity/kargo/pull/1515/commits/656c6154338b3edbe011ff93effe7deb53da9d14. It will also resolve https://github.com/akuity/kargo/pull/1515#discussion_r1500090417.